### PR TITLE
Purchase order

### PR DESCRIPTION
**Pull request by ike**
Purchase Order business object

### CODE REVIEW

File: lib/syspro/api_operations/transaction.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        end
+
+        private
+
+        def warn_on_opts_in_params(params)
+          Util::OPTS_USER_SPECIFIED.each do |opt|
+            if params.key?(opt)
+              warn("WARNING: #{opt} should be in opts instead of params.")
+            end
+          end
```
Comment (Samsinite) [inline lib/syspro/api_operations/transaction.rb]: Nothing wrong with this code, just wanted to point out a functional way of doing the same thing:
```
Util::OPTS_USER_SPECIFIED.select { |opt| params.key?(opt) }.each do |opt|
  warn(...)
end
```

File: lib/syspro/business_objects/parsers/portor_parser.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+          key = {}
+          key[:jnl_year] = doc.first_element_child.xpath('JnlYear')
+          key[:jnl_month] = doc.first_element_child.xpath('JnlMonth')
+          key[:journal] = doc.first_element_child.xpath('Journal')
+          key[:entry_number] = doc.first_element_child.xpath('EntryNumber')
+          key[:warehouse] = doc.first_element_child.xpath('Warehouse')
+          key[:gl_journal] = gl_journal_obj
+
+          binding.pry
```
Comment (jweakley) [inline lib/syspro/business_objects/parsers/portor_parser.rb]: You have a binding.pry

File: lib/syspro/business_objects/portor.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      include Syspro::ApiOperations::Transaction
+      include Syspro::BusinessObjects::Parsers
+
+      attr_accessor :transaction_date, :ignore_warnings, :non_stocked_wh_to_use, :grn_matching_action,
+        :allow_blank_supplier, :apply_if_entire_document_valid, :validate_only,
+        :manual_serial_transfers_allowed, :ignore_analysis
+
+      def call(user_id)
+        raise NotImplementedError.new("PORTOR not implemented yet.")
+        #xml_parameters = params_template.result(binding)
```
Comment (jweakley) [inline lib/syspro/business_objects/portor.rb]: Can we remove this comment block?

File: test/por_test.rb [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+  let(:username) { 'wland' }
+  let(:password) { 'piperita2016' }
+  let(:company) { 'L' }
+  let(:company_password) { '' }
+  let(:user_id) do
+    Syspro::Logon.logon(username, password, company, company_password)
+  end
+
+  #def test_por_transaction
```
Comment (jweakley) [inline test/por_test.rb]: Lets remove this

File: lib/syspro/business_objects/portor.rb (hunk 3 above)
Comment (ike) [reply lib/syspro/business_objects/portor.rb]: Need to implement this later, so I would rather leave it.

File: lib/syspro/api_operations/transaction.rb (hunk 1 above)
Comment (ike) [reply lib/syspro/api_operations/transaction.rb]: Cool! Thanks

**Pull request merged**